src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -294,9 +294,12 @@ public class ArgumentsCore
                 ShortName = "size")]
             public int? InputSize;
 
-            [Argument(ArgumentType.AtMostOnce, HelpText = "Source column separator. Options: tab, space, comma, single character", ShortName = "sep")]
+            [Argument(ArgumentType.AtMostOnce, Visibility = ArgumentAttribute.VisibilityType.CmdLineOnly, HelpText = "Source column separator. Options: tab, space, comma, single character", ShortName = "sep")]
             public string Separator = "tab";
 
+            [Argument(ArgumentType.AtMostOnce, Visibility = ArgumentAttribute.VisibilityType.EntryPointsOnly, HelpText = "Source column separator. Option: single character ONLY", ShortName = "del")]
```

Comment:
>Argument [](start = 13, length = 8)

This is a little unfortunate... on the one hand we have `Separator`. On the other hand we have `Delimiter`. I know you're doing this to avoid a name collision, but it would be nice if, from the *user's* perspective, the same concept had the same name in API usage and command line usage.

I think the `Name` attribute on `Argument` exists for this purpose. You can name the field whatever (maybe `Delimiter` could instead be `SeparatorChar`, or `SeparatorChars` if you fix it to be `char[]` instead). Then you can use `Name = nameof(Separator)` to make sure it has the same name in command line usage. The shortname I hope can I think be consistent between the two as well.